src/main/java/com/ericsson/gerrit/plugins/eiffel/handlers/StateHandler.java

```diff
@@ -0,0 +1,187 @@
+package com.ericsson.gerrit.plugins.eiffel.handlers;
+
+import java.io.File;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.ericsson.gerrit.plugins.eiffel.events.EiffelEvent;
+import com.google.gson.Gson;
+
+public class StateHandler {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(StateHandler.class);
+    private static final String FILE_ENDING = "db";
+    private final File pluginDir;
+
+    /**
+     * Constructor for the StateHandler class. in case the stateHandler is initiated
+     * with empty string or null the StateHandeling will be disabled.
+     *
+     * @param pluginDir
+     */
+    public StateHandler(final File pluginDir) {
+        this.pluginDir = pluginDir;
+    }
+
+    /**
+     * This function returns the last set scm change eiffel event id stored based on
+     * the branch and specific project.
+     *
+     * @param project
+     * @param branch
+     * @return
+     * @throws SomeRunTimeException
+     */
+    public String getLastSourceChangeSubmittedEiffelEvent(String project, String branch) throws NoSuchElementException {
+        return getLastCreatedEiffelEvent(project, branch, Table.SCS_TABLE);
+    }
+
+    public void setState(String project, String branch, String changeId, EiffelEvent eiffelEvent) {
```

Comment:
Do we need to save the whole Eiffel event? the DB files would grow in size quickly. all we need to save is the EventIds